app/code/Magento/ConfigurableProduct/Pricing/Price/LowestPriceOptionsProvider.php

```diff
@@ -41,34 +47,38 @@ class LowestPriceOptionsProvider implements LowestPriceOptionsProviderInterface
      * @param ResourceConnection $resourceConnection
      * @param LinkedProductSelectBuilderInterface $linkedProductSelectBuilder
      * @param CollectionFactory $collectionFactory
+     * @param StoreManagerInterface $storeManager
      */
     public function __construct(
         ResourceConnection $resourceConnection,
         LinkedProductSelectBuilderInterface $linkedProductSelectBuilder,
-        CollectionFactory $collectionFactory
+        CollectionFactory $collectionFactory,
+        StoreManagerInterface $storeManager
     ) {
         $this->resource = $resourceConnection;
         $this->linkedProductSelectBuilder = $linkedProductSelectBuilder;
         $this->collectionFactory = $collectionFactory;
+        $this->storeManager = $storeManager;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getProducts(ProductInterface $product)
     {
-        if (!isset($this->linkedProductMap[$product->getId()])) {
+        $key = $this->storeManager->getStore()->getId() . '-' . $product->getId();
```

Comment:
Price of product in Magento can be modified only on website level, so better use website id here

Reply:
You are right, prices and tax class id can be modified only on website level, but scope for special prices from and special prices to can be changed from admin panel http://prntscr.com/in1pba . So that attribute can affect results.

Reply:
I believe that scope for 'Special Price From Date' is wrong. All price related option should be a website or global level.

Reply:
You are totally right, but we have what we have and that works fine. If we will change for website level - in environmen emulation we will get different results than on real website. That why i propose to keep it without changes. 


I tried to create 3 products: 2 simple, 1 configurable and set 'Special Price From Date'  for store view level. 
Results:
Default store view: http://prntscr.com/ipevz0
Second store view: http://prntscr.com/ipewma

Product settings: 
http://prntscr.com/ipex7e (Second store view)
http://prntscr.com/ipexld (all stores and default store view is without overrides)

Reply:
On my opinion, store view level caching shouldn't affect performance.

If i remember correctly 'Special Price From Date' attribute doesn't exist in price index, that why it could be in store view level and works correctly.

Reply:
The last word is yours ;)

Reply:
In any case, your solution fixes the problem and improves the system, and my comment is just a suggestion for improvement, you can ignore it as a minor